### PR TITLE
Add remove function

### PR DESCRIPTION
**Pull request by emarkees**
Used GitHub Flow
✅ Used linters
✅ Used basic JavaScript syntax
✅ Used JavaScript to manipulate DOM elements
✅ Used JavaScript events
✅ Used objects to store and access data
✅ Documented my work in a professional way
✅ Worked with proper ES6 syntax
✅ Used npm as a software packages system
✅ Used webpack to bundle JavaScript
✅ Used ES6 modules to write modular JavaScript

### CODE REVIEW

File: src/print.js [hunk 1]
```diff
@@ -1,42 +1,76 @@
...
+      'beforeend',
+      `<ul class="listcontainer ">
       <div class="checBox">
-        <input type="checkbox" class="k">
+        <input type="checkbox" class="checkboxClass" id="myCheckbox">
       </div>
-      <li class="list textTodo">${item.description}</li><i class="fa-solid fa-ellipsis-vertical vertical" ></i>
+      <li class="list textTodo">${item.list}</li>
+      <i class="fa-solid fa-ellipsis-vertical vertical" ></i>
+      <i class="fa-solid fa-trash-can delete" id="${item.list}"></i>
```
Comment (Shisui6) [inline src/print.js]: According to the project requirements, new tasks should have the property completed set to false and the property index set to the value of the new array length. This feature is not yet implemented in your code. Kindly update the function for adding a new task🙏
image

image

File: src/print.js [hunk 2]
```diff
@@ -1,42 +1,76 @@
...
+    }
+  }
 
-let html = '';
+  // Define a remove method to remove a todo item from the DOM and the todoLis array
+  remove = (idSelected) => {
+    const { parentElement } = document.getElementById(idSelected);
+    if (parentElement) {
+      parentElement.remove();
+    }
```
Comment (Shisui6) [inline src/print.js]: According to the project requirements, deleting a task should update the index property of the remaining items so they represent the current list order and are unique. Your tasks should have three properties (index, description, and completed). These properties should be updated when tasks are added and removed from the list. Please update your to-do list to implement these requirements🙏

image

File: src/print.js [hunk 3]
```diff
@@ -1,42 +1,76 @@
...
-  html += `
-  <div>
-    <ul class="listcontainer ">
+    const indexToRemove = this.todoList.findIndex((e) => e.list === idSelected);
+    if (indexToRemove > -1) {
+      this.todoList.splice(indexToRemove, 1);
+      localStorage.setItem('todoList', JSON.stringify(this.todoList));
+    }
+  };
+
```
Comment (Shisui6) [inline src/print.js:45]: According to the project requirements, you should also implement a function for editing task descriptions. This functionality is not yet implemented in your code. Please update your to-do list to include this function🙏

image

**Review by Shisui6 — CHANGES_REQUESTED** [review]
Hi @emarkees, 

# Requires Changes 🍀
You have done a great job so far!👏👏
You are almost there, there are just a few thing to work on🪲

## Highlights
- Add and remove functions created ✅
- Task info saved in local storage ✅


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/print.js [hunk 4]
```diff
@@ -1,42 +1,179 @@
...
+    /*  const $editButton = document.getElementById(`edit-${item.description}`);
+    $editButton.addEventListener('click', () => {
+      const $inputField = document.getElementById('input'); // input div
+      $inputField.value = item.description;
+      $inputField.addEventListener('change', () => {
+        const newDescription = $inputField.value;
+        const indexToEdit = this.todoList.findIndex((e) => e.description === item.description);
+        if (indexToEdit > -1) {
+          this.edit(indexToEdit, newDescription);
+          /* this.updateIndexes();
```
Comment (AnselemOdims) [inline src/print.js]: - [ ] Please kindly remove all forms of commented code from your codebase as it pollutes it
- [ ] Also, the edit functionality is not working. Please kindly implement it.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @emarkees , 
Good job implementing some of the changes requested by the previous reviewer.

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/print.js [hunk 5]
```diff
@@ -1,42 +1,134 @@
...
+    if (todoToEdit) {
+      todoToEdit.description = newDescription;
+      const $todoDescription = document.querySelector(`.textTodo[data-index="${indexToEdit}"]`);
+      if ($todoDescription) {
+        $todoDescription.textContent = newDescription;
+        this.updateIndexes();
+      }
+      localStorage.setItem('todoList', JSON.stringify(this.todoList));
+    }
+  };
```
Comment (danmainah) [inline src/print.js:134]: - [ ] @emarkees Great work implementing the changes requested however the edit future still doesn't work, and there is no way to edit the todos, kindly ensure your edit functionality is working as previously requested.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @emarkees , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/print.js (hunk 5 above)
Comment (emarkees) [reply src/print.js:134]: @danmainah The future is working from my end, I will upload a clip of it,

**Review by DeliceLydia — APPROVED** [review]
## Hi @emarkee, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Edit feature was addressed is working


### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**